### ISSUE TITLE
Another error when run theme-check

### ISSUE THREAD

**Issue by pebiantara** [Shopify/theme-check#433]
Hey there,

previously i was reported an issue https://github.com/Shopify/theme-check/issues/428, still get an error but looks different error, is this related with theme-check version? because i was used theme-check and is working properly, then when i install theme-check again on another computer i faced this issues.

Checking build ...
Exception while running `SpaceInsideBraces#on_tag`:
```
ArgumentError: Bad start_index
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/position.rb:112:in `initialize'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/position.rb:19:in `new'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/position.rb:19:in `initialize'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/offense.rb:63:in `new'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/offense.rb:63:in `initialize'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/check.rb:95:in `new'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/check.rb:95:in `add_offense'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/checks/space_inside_braces.rb:61:in `on_tag'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/checks.rb:42:in `block in call_check_method'
  /usr/local/Cellar/ruby/3.0.2/lib/ruby/3.0.0/timeout.rb:97:in `block in timeout'
  /usr/local/Cellar/ruby/3.0.2/lib/ruby/3.0.0/timeout.rb:35:in `block in catch'
  /usr/local/Cellar/ruby/3.0.2/lib/ruby/3.0.0/timeout.rb:35:in `catch'
  /usr/local/Cellar/ruby/3.0.2/lib/ruby/3.0.0/timeout.rb:35:in `catch'
  /usr/local/Cellar/ruby/3.0.2/lib/ruby/3.0.0/timeout.rb:112:in `timeout'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/checks.rb:41:in `call_check_method'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/checks.rb:11:in `block in call'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/checks.rb:10:in `each'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/checks.rb:10:in `call'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:35:in `call_checks'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:22:in `visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `block in visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `each'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `block in visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `each'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `block in visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `each'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `block in visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `each'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:24:in `visit'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/visitor.rb:12:in `visit_template'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/analyzer.rb:39:in `block (2 levels) in analyze_theme'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/analyzer.rb:38:in `each'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/analyzer.rb:38:in `block in analyze_theme'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check.rb:57:in `with_liquid_c_disabled'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/analyzer.rb:37:in `analyze_theme'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/cli.rb:183:in `check'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/cli.rb:114:in `run!'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/cli.rb:118:in `run'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/lib/theme_check/cli.rb:140:in `parse_and_run'
  /usr/local/Cellar/theme-check/1.5.0/gems/theme-check-1.5.0/exe/theme-check:6:in `<top (required)>'
  /usr/local/bin/theme-check:6:in `load'
  /usr/local/bin/theme-check:6:in `<main>'
```

Template: `layout/checkout.liquid`
Node: `Liquid::Assign`
Markup:
```
assign discountCode = ""
```
Check options: `{}
`
Theme Check Version: 1.5.0
Ruby Version: 3.0.2
Platform: x86_64-darwin19
Muffin mode: activated

**Comment by charlespwd** [comment]
Looks like a duplicate of #420. 

Seems to be caused by having {%- on one side and %} on the other (or vice versa). Bugfix incoming (see #423)

**Comment by charlespwd** [comment]
Fix incoming to 1.5.2 (should be released today)
